modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/address/storage/AddressStorage.scala

```diff
@@ -0,0 +1,9 @@
+package org.tessellation.dag.l1.domain.address.storage
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+
+trait AddressStorage[F[_]] {
+  def getBalance(address: Address): F[Balance]
+  def updateBalances(addressBalances: Map[Address, Balance]): F[Unit]
```

Comment:
I'd make two separate functions:
```
setBalance(address, balance)
```
and
```
transfer(source, destination)
```

The latter will allow using storage (or db) locks to make the transfer transactional

Reply:
Currently I only need the transactional batch update. Hence `updateBalances` function. I could extract part of its implementation to  private `setBalance` that would return Quills abstraction `Free[ConnectionOp, Unit]` that could be reused once we need to set balance for a single address transactionally. Unfortunately I can't write transactional function for single transfer and then execute it again transactionally for multiple transfers. And I want either all to happen here or none here so I can't execute sequential transactional updates here.

Reply:
Ok, got it. You want to update balances for the whole block or snapshot at once. I thought about accepting single txs. If so, it's not needed - you can ignore my comment ;)